dvc/exceptions.py

```diff
@@ -261,6 +261,15 @@ def __init__(self, target_infos, stats=None):
                 "\n".join(targets), error_link("missing-files"),
             )
         )
+
+        if failed_not_stored:
+            prepend = (
+                "The following files are marked as `store: false` so the "
+                "remote was not searched, but are not present in local "
+                "cache:\n{}".format("\n".join(failed_not_stored))
+            )
+            m = prepend + "\n\n" + m
```

Comment:
```suggestion
            m = (
                "The following outputs are marked as `store: false` so "
                "remote storage was disabled, and are not present in the "
                "cache:\n{}\n\n".format("\n".join(failed_not_stored))
            ) + m
```
> May need wrapping?

Reply:
And how about also introducing a link to somewhere in the docs where this will be explained? (In anticipation of iterative/dvc.org/pull/1788)

Reply:
```
ERROR: failed to pull data from the cloud - The following files...
no_store.txt

Checkout failed for following targets:
no_store.txt
```

Those 2 new lines and repeated list items may be confusing though. Not sure how to improve the UI here and it's not a blocking problem but would be nice to try 🙂

Reply:
I've added a docs link like this:
```
The following outputs are marked as `store: false` so remote storage was disabled, and are not present in the cache:
foo
<https://error.dvc.org/checkout-failed-no-store>
Checkout failed for following targets:
foo
Is your cache up to date?
<https://error.dvc.org/missing-files>
``` 
and a section to the docs in https://github.com/iterative/dvc.org/pull/1788.